homeassistant/components/modbus/sensor.py

```diff
@@ -141,11 +143,11 @@ async def async_setup_platform(
             _LOGGER.error("Error in sensor %s structure: %s", entry[CONF_NAME], err)
             continue
 
-        if entry[CONF_COUNT] * 2 != size:
+        if entry[CONF_DATA_SIZE] != size:
             _LOGGER.error(
-                "Structure size (%d bytes) mismatch registers count (%d words)",
+                "Structure size (%d bytes) mismatch registers size (%d bytes)",
                 size,
-                entry[CONF_COUNT],
+                entry[CONF_DATA_SIZE],
```

Comment:
That needs to be related to count. Your suggestion breaks existing configurations.

Reply:
Have a default

Only above 2 bytes will break

But you dont support yet

With this custom is fine

No need for other datatypes

Reply:
count: 4 will break, look at the test cases that is supported and tested.

apart from that you break current valid configurations and that is not good.